parse/parser.go

```diff
@@ -104,11 +104,16 @@ func expectNext(buf TokenBuffer, t TokenType) (bool, error) {
 }
 
 func curPrecedence(buf TokenBuffer) precedence {
-	return precedenceMap[buf.Peek(CURRENT).Type]
+	if p, ok := precedenceMap[buf.Peek(CURRENT).Type]; ok {
```

Comment:
I think this is not needed. first checking existence is quite troublesome... and second if value don't exist than map automatically return 0 and this can be working as LOWEST

Reply:
> I think this is not needed. first checking existence everytime is quite troublesome... and second if value don't exist than map automatically return 0 and this can be working as LOWEST

LOWEST is 1.. So I added this code. But, i think there are other ways like adding Int, String... in precedence map.

Reply:
ok on second thought checking map value might be good